PIL/GifImagePlugin.py

```diff
@@ -602,79 +615,64 @@ def _get_palette_bytes(im, palette, info):
         else:
             source_palette = bytearray([i//3 for i in range(768)])
 
-    used_palette_colors = palette_bytes = None
+    palette_bytes = None
 
-    if _get_optimize(im, info):
-        used_palette_colors = _get_used_palette_colors(im)
+    used_palette_colors = _get_optimize(im, info)
+    if used_palette_colors is not None:
+        palette_bytes = b""
```

Comment:
I think this block should be hoisted into a separate method -- optimize_palette or something similar.